pkg/pusher/pusher.go

```diff
@@ -103,6 +105,12 @@ func (s *Service) chunksWorker() {
 				continue
 			}
 
+			if t != nil {
+				// update the tags only if we get it
+				t.Inc(tags.StateSent)
+				s.addUidToPushed(ch.Address().String(), t.Uid)
+			}
+
 			// set chunk status to synced
 			s.setChunkAsSynced(ctx, ch.Address())
```

Comment:
why don't you just add another parameter `*uint32` to `setChunkAsSynced` instead of adding a new field to the struct? and the mutex, and getters, delete, etc, wouldn't it be cleaner to just add one parameter here?

Reply:
> why don't you just add another parameter *uint32 to setChunkAsSynced instead of adding a new field to the struct? and the mutex, and getters, delete, etc, wouldn't it be cleaner to just add one parameter here?

@acud We were talking about making the pusher asyc right? we said it blocks until it gets receipt and does only 5 req/sec. That's the reason i moved to the old model of storing the chunk until we get a receipt. Anyway i will move to the model as you are suggesting now and come back to this when we do async.


>statesynced should be incremented after the receipt is received.

@Zelig True for intermediate nodes. But it is true for the sending node also.  Should we increment the tag after we set the ModeSetSyncPush in DB without errors? This is tricky to do. Fo now i would suggest i set the tag as soon as we receive receipt in all places. WDYS?

Reply:
> for intermediate nodes

intermediate nodes should not do any tag operations.

Reply:
Then all the tags operation should be in pusher... not in protocol.

Reply:
> intermediate nodes should not do any tag operations.

@acud 
https://github.com/ethersphere/bee/pull/330/files PR removed what you you are referring. 
1)  If you are sending chunks async, we needed a a place to store them till we get receipt. 
2) if you should not store tags in intermediate nodes, then we should store them as i did before   in pusher.

What i fail to understand is why tags was removed in that PR altogether !!!!